maven/src/main/java/org/commonjava/maven/galley/maven/model/view/MavenPomView.java

```diff
@@ -119,7 +121,8 @@ public MavenPomView( final ProjectVersionRef ref, final List<DocRef<ProjectVersi
         return new HashSet<String>( resolveXPathExpressionToAggregatedList( "/profiles/profile/id/text()", false, -1 ) );
     }
 
-    public String resolveMavenExpression( final String expression, final String... activeProfileIds )
+    public String resolveMavenExpression( final String expression, final RecursionInterceptor ri,
```

Comment:
Sorry to be picky, but this is public API, and we're going to need to maintain the old method signature. I'd suggest making the old method signature into a new method that delegates to this one with a null ri, then let it be instantiated down the call stack in resolveExpressions(..). This way we preserve backward compatibility while also fixing the property substitution.